File: app/src/main/java/ch/patland/loopyloop/SwipeToDeleteCallback.java
package ch.patland.loopyloop;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.ItemTouchHelper;
import androidx.recyclerview.widget.RecyclerView;

abstract class SwipeToDeleteCallback extends ItemTouchHelper.Callback {

    @Override
    public int getMovementFlags(@NonNull RecyclerView recyclerView,
                                         @NonNull RecyclerView.ViewHolder viewHolder) {
        int swipeFlag = ItemTouchHelper.LEFT | ItemTouchHelper.RIGHT;
        return makeMovementFlags(0, swipeFlag);
    }

    @Override
    public boolean onMove(
            @NonNull RecyclerView recyclerView,
            @NonNull RecyclerView.ViewHolder viewHolder,
            @NonNull RecyclerView.ViewHolder target
    ) {
        return false;
    }
}

File: app/src/main/java/ch/patland/loopyloop/SwipeToDeleteCallback.kt
package ch.patland.loopyloop;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.ItemTouchHelper;
import androidx.recyclerview.widget.RecyclerView;

abstract class SwipeToDeleteCallback: ItemTouchHelper.Callback() {

    @Override
    override fun getMovementFlags(@NonNull recyclerView: RecyclerView,
        @NonNull viewHolder: RecyclerView.ViewHolder): Int {
        val swipeFlag = ItemTouchHelper.LEFT or ItemTouchHelper.RIGHT
        return makeMovementFlags(0, swipeFlag);
    }

    @Override
    override fun onMove(
            @NonNull recyclerView: RecyclerView,
            @NonNull viewHolder: RecyclerView.ViewHolder,
            @NonNull target: RecyclerView.ViewHolder
    ): Boolean {
        return false;
    }
}
